assets/src/modules/BaseLayersMap.js

```diff
@@ -0,0 +1,232 @@
+                    title: 'Bing Road',
+                    imagerySet: 'RoadOnDemand'
+                },
+                bingSatellite : {
+                    title: 'Bing Aerial',
+                    imagerySet: 'Aerial'
+                },
+                bingHybrid : {
+                    title: 'Bing Hybrid',
+                    imagerySet: 'AerialWithLabelsOnDemand'
+                }
+            };
+
+            for (const key in bingConfigs) {
+                if(mainLizmap.config.options?.[key]){
+
+                    const bingConfig = bingConfigs[key];
+
+                    this.addLayer(
+                        new TileLayer({
+                            title: bingConfig.title,
+                            preload: Infinity,
+                            source: new BingMaps({
+                                key: mainLizmap.config.options.bingKey,
+                                imagerySet: bingConfig.imagerySet,
+                                culture: navigator.language
+                            }),
+                        })
+                    );
+                }
+            }
+        }
+
+        this._baseLayers = [];
+        let firstBaseLayer = true;
+        let baseLayers = [];
+        if(mainLizmap.config?.baseLayers){
+            baseLayers = Object.entries(mainLizmap.config.baseLayers);
+        }
+        for (const [title, params] of baseLayers) {
+            if(params.type = 'xyz'){
+                this._baseLayers.push(
+                    new TileLayer({
+                        title: title,
+                        visible: firstBaseLayer,
+                        source: new XYZ({
+                            url: params.url,
+                            minZoom: params?.zmin,
+                            maxZoom: params?.zmax,
+                        })
+                    })
+                );
+            }
+
+            firstBaseLayer = false;
+        }
+
+        const layerGroup = new LayerGroup({
+            layers: this._baseLayers
+        });
+
+        layerGroup.on('change', () => {
+            mainEventDispatcher.dispatch('baseLayers.changed');
+        });
+
+        this.setLayerGroup(layerGroup);
+
+        // Overlay layers
+        for (const [title, params] of Object.entries(mainLizmap.config?.layers)) {
+            let extent = params.extent;
+            if(params.crs !== mainLizmap.projection){
+                extent = transformExtent(extent, params.crs, mainLizmap.projection);
+            }
+            const minResolution = Utils.getResolutionFromScale(params.minScale);
+            const maxResolution = Utils.getResolutionFromScale(params.maxScale);
+
+            if (params.cached === "False") {
+                this.addLayer(new ImageLayer({
+                    extent: extent,
+                    minResolution: minResolution,
+                    maxResolution: maxResolution,
+                    visible: params.toggled === "True",
```

Comment:
Can you make a global function to parse either a string or bool and return its boolean value ? 
Similar to https://github.com/3liz/lizmap-plugin/blob/master/lizmap/tools.py#L180...L191

It would be nice to close your own ticket :)
https://github.com/3liz/lizmap-plugin/issues/176
It's one line of code in lizmap plugin to switch (I can make the switch when the server is targeting > LWC 3.7

Side note, the comparaison L179 looks impossible after L174 ?

Reply:
We talked with @mdouchin having the correct cfg sent by the server in Lizmap 3.7. I'll then have to replace all "True"/"False" tests.

Reply:
I mean calling a global function like the existing one https://github.com/3liz/lizmap-web-client/blob/78b57ac29f67880a12d5072b00bf422b59f67e58/lizmap/www/assets/js/dataviz/dataviz.js#L85